src/types/dvalue.rs

```diff
@@ -0,0 +1,200 @@
+use crate::error::ActiveStorageError;
+
+/// A value of any DType.
+/// This is an alias of the Number type from serde_json, which is an enum that can represent
+/// integers and floating point numbers.
+/// The number type is an enum over i64, u64 and f64, with the additional constraint that floating
+/// point numbers must be finite (not positive or negative infinity or NaN).
+pub type DValue = serde_json::Number;
+
+/// Try to convert a DValue to an i64.
+fn as_i64(value: &DValue) -> Result<i64, ActiveStorageError> {
+    value
+        .as_i64()
+        .ok_or(ActiveStorageError::IncompatibleMissing(value.clone()))
+}
+
+/// Try to convert a DValue to a u64.
+fn as_u64(value: &DValue) -> Result<u64, ActiveStorageError> {
+    value
+        .as_u64()
+        .ok_or(ActiveStorageError::IncompatibleMissing(value.clone()))
+}
+
+/// Try to convert a DValue to an f64.
+fn as_f64(value: &DValue) -> Result<f64, ActiveStorageError> {
+    value
+        .as_f64()
+        .ok_or(ActiveStorageError::IncompatibleMissing(value.clone()))
+}
+
+/// Attempt to convert from a [DValue](crate::types::DValue) to specific numeric type.
+// This trait exists because we can't implement TryFrom<DValue> for numeric types because the trait
+// and type are in external crates.
+pub trait TryFromDValue: Sized {
+    /// Try to convert from a [DValue](crate::types::DValue) to a numeric type.
+    fn try_from_dvalue(value: DValue) -> Result<Self, ActiveStorageError>;
+}
```

Comment:
Did you consider using the [new type pattern](https://doc.rust-lang.org/book/ch19-03-advanced-traits.html#using-the-newtype-pattern-to-implement-external-traits-on-external-types) to define `DValue` from `serde_json::Number` so that you could then implement external traits on `DValue`?

Reply:
I did briefly try this, but found that you lose the ability to use methods and traits associated with the wrapped type, which is a bit inconvenient.